bin/generate.py

```diff
@@ -245,46 +245,55 @@ def run_interactive(self, problem, bar, cwd, t):
 
 
 # Return absolute path to default submission, starting from the submissions directory.
-# This function will always raise a warning.
+# This function will always prints a message.
 # Which submission is used is implementation defined, unless one is explicitly given on the command line.
 def default_solution_path(generator_config):
     problem = generator_config.problem
+    solution = None
+    stored_solution = problem.tmpdir / '.default_solution'
     if config.args.default_solution:
-        solution = problem.path / config.args.default_solution
         if generator_config.has_yaml:
             message(
-                f'''Prefer setting the default solution permanently:
-solution: /{solution.relative_to(problem.path)}''',
+                f'''--default-solution Ignored. Set the default solution in the generator.yaml!
+solution: /{config.args.default_solution}''',
                 'generators.yaml',
-                color_type=MessageType.LOG,
+                color_type=MessageType.WARN,
             )
+        else:
+            solution = problem.path / config.args.default_solution
     else:
         # Use one of the accepted submissions.
         solutions = list(glob(problem.path, 'submissions/accepted/*'))
         if len(solutions) == 0:
             fatal(f'No solution specified and no accepted submissions found.')
-        if generator_config.has_yaml:
-            # Note: we explicitly random shuffle the submission that's used to generate answers to
-            # encourage setting it in generators.yaml.
+
+        # always try to take the same solution to not mess with hashing
+        if stored_solution.is_file():
```

Comment:
This is only needed when `!generator_config.has_yaml`, right? So can be moved into the `else:` at line 290 below.

Reply:
I want to pick the same solution again even when the user deletes the line from the `generator.yaml`

Reply:
ah right sure; feels a bit redundant since you agressively add it back in but doesn't really hurt I suppose :)